domains/auth/Auth.API/Unit.Tests/Controllers/OidcControllerTests.cs

```diff
@@ -58,698 +64,563 @@ public OidcControllerTests()
+                },
+                new List<string>{"TestRoleKeyAdmin","TestRoleKeyAdmin2","TestRoleKeyAdmin3","TestRoleKeyAdmin4"}
             },
-            KeySetUsing(tokenOptions.PublicKeyPem)
-        );
-
-        cache.GetAsync().Returns(document);
-
-        var providerId = Guid.NewGuid().ToString();
-        var name = Guid.NewGuid().ToString();
-        var identityToken = TokenUsing(tokenOptions, identity, oidcOptions.ClientId);
-        var accessToken = TokenUsing(tokenOptions, access, oidcOptions.ClientId, claims: new() {
-            { "scope", "something" },
-        });
-        var userToken = TokenUsing(tokenOptions, user, oidcOptions.ClientId, claims: new() {
-            { "mitid.uuid", providerId },
-            { "mitid.identity_name", name }
-        });
-
-        http.When(HttpMethod.Post, tokenEndpoint.AbsoluteUri).Respond("application/json", $$"""{"access_token":"{{accessToken}}", "id_token":"{{identityToken}}", "userinfo_token":"{{userToken}}"}""");
-        factory.CreateClient(Arg.Any<string>()).Returns(http.ToHttpClient());
-
-        var result = await new OidcController().CallbackAsync(metrics, cache, factory, userProviderService, service, cryptography, issuer, oidcOptions, providerOptions, roleOptions, logger, Guid.NewGuid().ToString(), null, null);
-
-        Assert.NotNull(result);
-        Assert.IsType<RedirectResult>(result);
-        var redirectResult = (RedirectResult)result;
-        var query = HttpUtility.UrlDecode(new Uri(redirectResult.Url).Query);
-        Assert.Contains($"{ErrorCode.QueryString}={ErrorCode.Authentication.InvalidTokens}", query);
-    }
+        };
 
     [Theory]
-    [InlineData("access_denied", "internal_error", ErrorCode.AuthenticationUpstream.InternalError)]
-    [InlineData("access_denied", "user_aborted", ErrorCode.AuthenticationUpstream.Aborted)]
-    [InlineData("access_denied", "private_to_business_user_aborted", ErrorCode.AuthenticationUpstream.Aborted)]
-    [InlineData("access_denied", "no_ctx", ErrorCode.AuthenticationUpstream.NoContext)]
-    [InlineData("access_denied", null, ErrorCode.AuthenticationUpstream.Failed)]
-    [InlineData("invalid_request", null, ErrorCode.AuthenticationUpstream.InvalidRequest)]
-    [InlineData("unauthorized_client", null, ErrorCode.AuthenticationUpstream.InvalidClient)]
-    [InlineData("unsupported_response_type", null, ErrorCode.AuthenticationUpstream.InvalidRequest)]
-    [InlineData("invalid_scope", null, ErrorCode.AuthenticationUpstream.InvalidScope)]
-    [InlineData("server_error", null, ErrorCode.AuthenticationUpstream.InternalError)]
-    [InlineData("temporarily_unavailable", null, ErrorCode.AuthenticationUpstream.InternalError)]
-    [InlineData(null, null, ErrorCode.AuthenticationUpstream.Failed)]
-    public async Task CallbackAsync_ShouldReturnRedirectToFrontendWithErrorAndLogWarning_WhenGivenErrorConditions(string? error, string? errorDescription, string expected)
+    [MemberData(nameof(RoleMatcherData))]
```

Comment:
Fact or Theory, you must choose only one... :)

Reply:
i have chosen